download/upgradeRecipe/upgradeRecipe8.adoc

```diff
@@ -50,4 +50,176 @@ Any backwards incompatible changes are annotated with a [.label.label-danger.lab
 Due to the limited usage of OSGi and the maintenance burden it brings,
 the OptaPlanner jars in the 8.x series no longer include OSGi metadata in their `META-INF/MANIFEST.MF` file.
 
+[.upgrade-recipe-major]
+=== `filterClassList` replaced by a single filterClass
 
+This change applies to a configuration of `EntitySelector`, `ValueSelector` and `MoveSelector` in both
+the configuration API and the solver configuration XML.
+
+==== Reason:
+In practice, you don't need multiple selection filter classes often, and you can always replace them by a single selection
+filter class which implements the logic of all of them.
+Passing a single selection class now requires less boilerplate code.
+
+Before in `*.java`:
+[source, java]
+----
+ValueSelectorConfig valueSelectorConfig = new ValueSelectorConfig();
+valueSelectorConfig.setFilterClassList(Collections.singletonList(MySelectionFilterClass.class));
+----
+
+After in `*.java`:
+[source, java]
+----
+ValueSelectorConfig valueSelectorConfig = new ValueSelectorConfig();
+valueSelectorConfig.setFilterClass(MySelectionFilterClass.class);
+----
+
+==== Replacing multiple selection filter classes with a single one
+
+Before in `\*.xml` and `*.java`:
+
+[source, xml]
+----
+<swapMoveSelector>
+  <entitySelector>
+    <filterClass>com.example.FilterA</filterClass>
+    <filterClass>com.example.FilterB</filterClass>
+  </entitySelector>
+</swapMoveSelector>
+----
+
+[source, java]
+----
+package com.example;
+...
+public class FilterA implements SelectionFilter<MySolution, MyPlanningEntity> {
+
+    @Override
+    public boolean accept(ScoreDirector<MySolution> scoreDirector, MyPlanningEntity selection) {
+        return selection.getValue() < 500;
+    }
+}
+----
+[source, java]
+----
+package com.example;
+...
+public class FilterB implements SelectionFilter<MySolution, MyPlanningEntity> {
+
+    @Override
+    public boolean accept(ScoreDirector<MySolution> scoreDirector, MyPlanningEntity selection) {
+        return selection.getOrder() == Order.ASC;
+    }
+}
+----
+After in `\*.xml` and `*.java`:
+
+[source, xml]
+----
+<swapMoveSelector>
+  <entitySelector>
+    <filterClass>com.example.SingleEntityFilter</filterClass>
+  </entitySelector>
+</swapMoveSelector>
+----
+
+[source, java]
+----
+package com.example;
+...
+public class SingleEntityFilter implements SelectionFilter<MySolution, MyPlanningEntity> {
+
+    @Override
+    public boolean accept(ScoreDirector<MySolution> scoreDirector, MyPlanningEntity selection) {
+        return selection.getValue() < 500 && selection.getOrder() == Order.ASC;
+    }
+}
+----
+
+[.upgrade-recipe-major]
```

Comment:
this should be minor, not major.

Major's are uncollapsed by default and should effect at least 20% of all users.
Minor's are collapsed by default and should have a title that allows users to easily figure out if it might apply to the feature subset that they use of optaplanner.

Reply:
Thanks for the clarification, I was not sure about the criteria for deciding what is a major and what is a minor change.